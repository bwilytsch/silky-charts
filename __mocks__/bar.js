/* ignore file coverage */

export const data = [
  { name: 'foo', value: 19 },
  { name: 'bar', value: 70 },
  { name: 'baz', value: 93 },
  { name: 'quz', value: 80 },
];
export const dataWidthDates = [
  { name: '2018-01-31T05:00:00.000Z', value: 19 },
  { name: '2018-02-01T05:00:00.000Z', value: 70 },
  { name: '2018-02-02T05:00:00.000Z', value: 93 },
  { name: '2018-02-03T05:00:00.000Z', value: 80 },
];
