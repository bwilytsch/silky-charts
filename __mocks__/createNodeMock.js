export default element => {
  if (element.type === 'svg') {
    return {
      parentElement: {},
    };
  }
};
