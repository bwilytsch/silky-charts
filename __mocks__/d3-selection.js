/* ignore file coverage */

const selection = jest.genMockFromModule('d3-selection');

const attr = () => ({ attr });

const fn = () => ({
  call: jest.fn(),
  select: jest.fn(() => ({ attr })),
  selectAll: jest.fn(() => ({ attr })),
});

selection.select = fn;
selection.selectAll = fn;

module.exports = selection;
