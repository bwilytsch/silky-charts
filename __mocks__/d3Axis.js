/* ignore file coverage */

module.exports = jest.fn(() => ({
  ticks: jest.fn(() => ({
    tickFormat: jest.fn(),
  })),
}));
