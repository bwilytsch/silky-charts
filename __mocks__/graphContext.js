/* ignore file coverage */

const rect = { top: 10, right: 10, bottom: 10, left: 10 };

module.exports = {
  margin: rect,
  node: { getBoundingClientRect: jest.fn(() => rect) },
  staticTooltip: false,
  visibleTicks: true,
  xAxisLabelRotation: false,
  xAxisLabelRotationValue: -65,
};
