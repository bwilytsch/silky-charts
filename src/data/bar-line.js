export const barLineData = [
  { name: 'January', apples: 1, bananas: 2, cherries: 3, dates: 16 },
  { name: 'February', apples: 11, bananas: 24, cherries: 6, dates: 15 },
  { name: 'March', apples: 13, bananas: 0, cherries: 11, dates: 3 },
  { name: 'April', apples: 20, bananas: 11, cherries: 18, dates: 19 },
  { name: 'May', apples: 8, bananas: 13, cherries: 3, dates: 1 },
  { name: 'June', apples: 22, bananas: 18, cherries: 25, dates: 0 },
  { name: 'July', apples: 17, bananas: 8, cherries: 21, dates: 1 },
  { name: 'August', apples: 23, bananas: 17, cherries: 13, dates: 18 },
  { name: 'September', apples: 5, bananas: 1, cherries: 2, dates: 19 },
  { name: 'October', apples: 19, bananas: 14, cherries: 9, dates: 12 },
  { name: 'November', apples: 7, bananas: 10, cherries: 18, dates: 8 },
  { name: 'December', apples: 12, bananas: 11, cherries: 6, dates: 1 },
];

export const barLineDates = [
  {
    name: '2018-12-30T05:00:00.000Z',
    apples: 0,
    bananas: 23,
    cherries: 11,
    dates: 17,
  },
  {
    name: '2018-12-31T05:00:00.000Z',
    apples: 4,
    bananas: 0,
    cherries: 10,
    dates: 12,
  },
  {
    name: '2019-01-01T05:00:00.000Z',
    apples: 6,
    bananas: 7,
    cherries: 7,
    dates: 10,
  },
  {
    name: '2019-01-02T05:00:00.000Z',
    apples: 16,
    bananas: 9,
    cherries: 20,
    dates: 9,
  },
  {
    name: '2019-01-03T05:00:00.000Z',
    apples: 0,
    bananas: 21,
    cherries: 24,
    dates: 21,
  },
  {
    name: '2019-01-04T05:00:00.000Z',
    apples: 17,
    bananas: 1,
    cherries: 12,
    dates: 8,
  },
  {
    name: '2019-01-05T05:00:00.000Z',
    apples: 12,
    bananas: 23,
    cherries: 2,
    dates: 7,
  },
  {
    name: '2019-01-06T05:00:00.000Z',
    apples: 21,
    bananas: 6,
    cherries: 21,
    dates: 21,
  },
  {
    name: '2019-01-07T05:00:00.000Z',
    apples: 13,
    bananas: 20,
    cherries: 21,
    dates: 7,
  },
];
