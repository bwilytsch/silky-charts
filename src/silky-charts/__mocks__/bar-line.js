/* ignore file coverage */

export const data = [
  { name: 'foo', apples: 18, bananas: 16, cherries: 17, dates: 1 },
  { name: 'bar', apples: 14, bananas: 14, cherries: 19, dates: 15 },
  { name: 'baz', apples: 17, bananas: 13, cherries: 18, dates: 9 },
  { name: 'qux', apples: 14, bananas: 13, cherries: 2, dates: 13 },
];
export const dataWidthDates = [
  {
    name: '2018-12-30T05:00:00.000Z',
    apples: 18,
    bananas: 16,
    cherries: 17,
    dates: 1,
  },
  {
    name: '2018-12-31T05:00:00.000Z',
    apples: 14,
    bananas: 14,
    cherries: 19,
    dates: 15,
  },
  {
    name: '2019-01-01T05:00:00.000Z',
    apples: 17,
    bananas: 13,
    cherries: 18,
    dates: 9,
  },
  {
    name: '2019-01-02T05:00:00.000Z',
    apples: 14,
    bananas: 13,
    cherries: 2,
    dates: 13,
  },
];
