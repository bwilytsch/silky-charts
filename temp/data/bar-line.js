export const barLineData = [
  { name: 'January', series: 'apples', value: 1 },
  { name: 'January', series: 'bananas', value: 2 },
  { name: 'January', series: 'cherries', value: 3 },
  { name: 'January', series: 'dates', value: 16 },
  { name: 'February', series: 'apples', value: 11 },
  { name: 'February', series: 'bananas', value: 24 },
  { name: 'February', series: 'cherries', value: 6 },
  { name: 'February', series: 'dates', value: 15 },
  { name: 'March', series: 'apples', value: 13 },
  { name: 'March', series: 'bananas', value: 0 },
  { name: 'March', series: 'cherries', value: 11 },
  { name: 'March', series: 'dates', value: 3 },
  { name: 'April', series: 'apples', value: 20 },
  { name: 'April', series: 'bananas', value: 11 },
  { name: 'April', series: 'cherries', value: 18 },
  { name: 'April', series: 'dates', value: 19 },
  { name: 'May', series: 'apples', value: 8 },
  { name: 'May', series: 'bananas', value: 13 },
  { name: 'May', series: 'cherries', value: 3 },
  { name: 'May', series: 'dates', value: 1 },
  { name: 'June', series: 'apples', value: 22 },
  { name: 'June', series: 'bananas', value: 18 },
  { name: 'June', series: 'cherries', value: 25 },
  { name: 'June', series: 'dates', value: 0 },
  { name: 'July', series: 'apples', value: 17 },
  { name: 'July', series: 'bananas', value: 8 },
  { name: 'July', series: 'cherries', value: 21 },
  { name: 'July', series: 'dates', value: 1 },
  { name: 'August', series: 'apples', value: 23 },
  { name: 'August', series: 'bananas', value: 17 },
  { name: 'August', series: 'cherries', value: 13 },
  { name: 'August', series: 'dates', value: 18 },
  { name: 'September', series: 'apples', value: 5 },
  { name: 'September', series: 'bananas', value: 1 },
  { name: 'September', series: 'cherries', value: 2 },
  { name: 'September', series: 'dates', value: 19 },
  { name: 'October', series: 'apples', value: 19 },
  { name: 'October', series: 'bananas', value: 14 },
  { name: 'October', series: 'cherries', value: 9 },
  { name: 'October', series: 'dates', value: 12 },
  { name: 'November', series: 'apples', value: 7 },
  { name: 'November', series: 'bananas', value: 10 },
  { name: 'November', series: 'cherries', value: 18 },
  { name: 'November', series: 'dates', value: 8 },
  { name: 'December', series: 'apples', value: 12 },
  { name: 'December', series: 'bananas', value: 11 },
  { name: 'December', series: 'cherries', value: 6 },
  { name: 'December', series: 'dates', value: 1 },
];

export const barLineDates = [
  { name: '2018-12-30T05:00:00.000Z', series: 'apples', value: 0 },
  { name: '2018-12-30T05:00:00.000Z', series: 'bananas', value: 23 },
  { name: '2018-12-30T05:00:00.000Z', series: 'cherries', value: 11 },
  { name: '2018-12-30T05:00:00.000Z', series: 'dates', value: 17 },
  { name: '2018-12-31T05:00:00.000Z', series: 'apples', value: 4 },
  { name: '2018-12-31T05:00:00.000Z', series: 'bananas', value: 0 },
  { name: '2018-12-31T05:00:00.000Z', series: 'cherries', value: 10 },
  { name: '2018-12-31T05:00:00.000Z', series: 'dates', value: 12 },
  { name: '2019-01-01T05:00:00.000Z', series: 'apples', value: 6 },
  { name: '2019-01-01T05:00:00.000Z', series: 'bananas', value: 7 },
  { name: '2019-01-01T05:00:00.000Z', series: 'cherries', value: 7 },
  { name: '2019-01-01T05:00:00.000Z', series: 'dates', value: 10 },
  { name: '2019-01-02T05:00:00.000Z', series: 'apples', value: 16 },
  { name: '2019-01-02T05:00:00.000Z', series: 'bananas', value: 9 },
  { name: '2019-01-02T05:00:00.000Z', series: 'cherries', value: 20 },
  { name: '2019-01-02T05:00:00.000Z', series: 'dates', value: 9 },
  { name: '2019-01-03T05:00:00.000Z', series: 'apples', value: 0 },
  { name: '2019-01-03T05:00:00.000Z', series: 'bananas', value: 21 },
  { name: '2019-01-03T05:00:00.000Z', series: 'cherries', value: 24 },
  { name: '2019-01-03T05:00:00.000Z', series: 'dates', value: 21 },
  { name: '2019-01-04T05:00:00.000Z', series: 'apples', value: 17 },
  { name: '2019-01-04T05:00:00.000Z', series: 'bananas', value: 1 },
  { name: '2019-01-04T05:00:00.000Z', series: 'cherries', value: 12 },
  { name: '2019-01-04T05:00:00.000Z', series: 'dates', value: 8 },
  { name: '2019-01-05T05:00:00.000Z', series: 'apples', value: 12 },
  { name: '2019-01-05T05:00:00.000Z', series: 'bananas', value: 23 },
  { name: '2019-01-05T05:00:00.000Z', series: 'cherries', value: 2 },
  { name: '2019-01-05T05:00:00.000Z', series: 'dates', value: 7 },
  { name: '2019-01-06T05:00:00.000Z', series: 'apples', value: 21 },
  { name: '2019-01-06T05:00:00.000Z', series: 'bananas', value: 6 },
  { name: '2019-01-06T05:00:00.000Z', series: 'cherries', value: 21 },
  { name: '2019-01-06T05:00:00.000Z', series: 'dates', value: 21 },
  { name: '2019-01-07T05:00:00.000Z', series: 'apples', value: 13 },
  { name: '2019-01-07T05:00:00.000Z', series: 'bananas', value: 20 },
  { name: '2019-01-07T05:00:00.000Z', series: 'cherries', value: 21 },
  { name: '2019-01-07T05:00:00.000Z', series: 'dates', value: 7 },
];
