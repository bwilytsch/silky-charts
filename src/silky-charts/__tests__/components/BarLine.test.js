import React from 'react';
import { BarLine } from '../../';
import { create } from 'react-test-renderer';
import 'jest-styled-components';
import { data, dataWidthDates } from '../../__mocks__/bar-line';

test('Should render correctly', () => {
  const tree = create(<BarLine data={data} />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Should render correctly with dates as name', () => {
  const tree = create(<BarLine data={dataWidthDates} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should render correctly with only bars', () => {
  const tree = create(
    <BarLine data={data} stackedKeys={['apples', 'bananas']} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Should render correctly with only lines', () => {
  const tree = create(
    <BarLine data={data} lineKeys={['cherries', 'dates']} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Should render correctly with bars and lines', () => {
  const tree = create(
    <BarLine
      data={dataWidthDates}
      stackedKeys={['apples', 'bananas']}
      lineKeys={['cherries', 'dates']}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Should render correctly with a different line type', () => {
  const tree = create(
    <BarLine data={data} lineType={'curveNatural'} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Should render correctly with curveBundle line type', () => {
  const tree = create(
    <BarLine data={data} lineType={'curveBundle'} lineTypeOption={0.5} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Should render correctly with curveBundle line type', () => {
  const tree = create(
    <BarLine data={data} lineType={'curveCardinalOpen'} lineTypeOption={10} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Should render correctly with curveBundle line type', () => {
  const tree = create(
    <BarLine
      data={data}
      lineType={'curveCatmullRomOpen'}
      lineTypeOption={0.5}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Should render correctly width horizontal grid', () => {
  const tree = create(<BarLine data={data} grid />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should render correctly width vertical grid', () => {
  const tree = create(<BarLine data={data} grid horizontal />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should render correctly X axis label rotation', () => {
  const tree = create(<BarLine data={data} xAxisLabelRotation />).toJSON();
  expect(tree).toMatchSnapshot();
});
