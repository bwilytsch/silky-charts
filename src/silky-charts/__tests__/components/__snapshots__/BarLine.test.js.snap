// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should remove resize event listener on unmount 1`] = `
.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-36"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;

exports[`Should remove resize event listener on unmount 2`] = `
<svg
  className="sc-bZQynM silky-charts cuUavZ"
  id="silky-charts-bar-line-36"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="sc-bdVaJa axis-x bDWFJH"
      transform="translate(0, 0)"
    />
    <g
      className="sc-bdVaJa axis-y bDWFJH"
    />
  </g>
</svg>
`;

exports[`Should render correctly 1`] = `
.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-0"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;

exports[`Should render correctly X axis label rotation 1`] = `
.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-34"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;

exports[`Should render correctly width horizontal grid 1`] = `
.c1 path {
  stroke: transparent;
}

.c1 line {
  stroke: rgb(220,220,220);
}

.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-30"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="grid c1"
    />
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;

exports[`Should render correctly width vertical grid 1`] = `
.c1 path {
  stroke: transparent;
}

.c1 line {
  stroke: rgb(220,220,220);
}

.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-32"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="grid c1"
    />
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;

exports[`Should render correctly with a different line type 1`] = `
.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-22"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;

exports[`Should render correctly with bars and lines 1`] = `
.c4 {
  cursor: pointer;
  fill: rgb(255,255,255);
  stroke: rgb(127,22,42);
  stroke-width: 3;
}

.c6 {
  cursor: pointer;
  fill: rgb(255,255,255);
  stroke: rgb(255,121,147);
  stroke-width: 3;
}

.c3 {
  fill: none;
  pointer-events: none;
  stroke: rgb(127,22,42);
  stroke-width: 3;
}

.c5 {
  fill: none;
  pointer-events: none;
  stroke: rgb(255,121,147);
  stroke-width: 3;
}

.c1 {
  fill: rgb(44,98,122);
  height: 0px;
  width: 0px;
}

.c1:hover {
  fill: hsl(198.5,47%,26%);
}

.c2 {
  fill: rgb(166,224,251);
  height: 0px;
  width: 0px;
}

.c2:hover {
  fill: hsl(199.10000000000002,91.4%,65.4%);
}

.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-20"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
    <g
      className="apples-layer"
    >
      <rect
        className="bar c1"
        color="rgb(44, 98, 122)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={18}
        x={0}
        y={0}
      />
      <rect
        className="bar c1"
        color="rgb(44, 98, 122)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={14}
        x={0}
        y={0}
      />
      <rect
        className="bar c1"
        color="rgb(44, 98, 122)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={17}
        x={0}
        y={0}
      />
      <rect
        className="bar c1"
        color="rgb(44, 98, 122)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={14}
        x={0}
        y={0}
      />
    </g>
    <g
      className="bananas-layer"
    >
      <rect
        className="bar c2"
        color="rgb(166, 224, 251)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={16}
        x={0}
        y={0}
      />
      <rect
        className="bar c2"
        color="rgb(166, 224, 251)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={14}
        x={0}
        y={0}
      />
      <rect
        className="bar c2"
        color="rgb(166, 224, 251)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={13}
        x={0}
        y={0}
      />
      <rect
        className="bar c2"
        color="rgb(166, 224, 251)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={13}
        x={0}
        y={0}
      />
    </g>
    <g
      className="cherries-layer"
    >
      <path
        className="line-path c3"
        color="rgb(127, 22, 42)"
        d="M0,0L0,0L0,0L0,0"
      />
      <g
        className="line-dot-group"
      >
        <circle
          className="line-dot c4"
          color="rgb(127, 22, 42)"
          cx={0}
          cy={0}
          r={5}
        />
        <circle
          className="line-dot c4"
          color="rgb(127, 22, 42)"
          cx={0}
          cy={0}
          r={5}
        />
        <circle
          className="line-dot c4"
          color="rgb(127, 22, 42)"
          cx={0}
          cy={0}
          r={5}
        />
        <circle
          className="line-dot c4"
          color="rgb(127, 22, 42)"
          cx={0}
          cy={0}
          r={5}
        />
      </g>
    </g>
    <g
      className="dates-layer"
    >
      <path
        className="line-path c5"
        color="rgb(255, 121, 147)"
        d="M0,0L0,0L0,0L0,0"
      />
      <g
        className="line-dot-group"
      >
        <circle
          className="line-dot c6"
          color="rgb(255, 121, 147)"
          cx={0}
          cy={0}
          r={5}
        />
        <circle
          className="line-dot c6"
          color="rgb(255, 121, 147)"
          cx={0}
          cy={0}
          r={5}
        />
        <circle
          className="line-dot c6"
          color="rgb(255, 121, 147)"
          cx={0}
          cy={0}
          r={5}
        />
        <circle
          className="line-dot c6"
          color="rgb(255, 121, 147)"
          cx={0}
          cy={0}
          r={5}
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`Should render correctly with curveBundle line type 1`] = `
.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-24"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;

exports[`Should render correctly with curveBundle line type 2`] = `
.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-26"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;

exports[`Should render correctly with curveBundle line type 3`] = `
.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-28"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;

exports[`Should render correctly with dates as name 1`] = `
.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-12"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;

exports[`Should render correctly with dates as name 2`] = `
.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-14"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;

exports[`Should render correctly with height 1`] = `
.c0 {
  height: 400px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-4"
  size={
    Object {
      "height": 400,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;

exports[`Should render correctly with only bars 1`] = `
.c1 {
  fill: rgb(44,98,122);
  height: 0px;
  width: 0px;
}

.c1:hover {
  fill: hsl(198.5,47%,26%);
}

.c2 {
  fill: rgb(166,224,251);
  height: 0px;
  width: 0px;
}

.c2:hover {
  fill: hsl(199.10000000000002,91.4%,65.4%);
}

.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-16"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
    <g
      className="apples-layer"
    >
      <rect
        className="bar c1"
        color="rgb(44, 98, 122)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={18}
        x={0}
        y={0}
      />
      <rect
        className="bar c1"
        color="rgb(44, 98, 122)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={14}
        x={0}
        y={0}
      />
      <rect
        className="bar c1"
        color="rgb(44, 98, 122)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={17}
        x={0}
        y={0}
      />
      <rect
        className="bar c1"
        color="rgb(44, 98, 122)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={14}
        x={0}
        y={0}
      />
    </g>
    <g
      className="bananas-layer"
    >
      <rect
        className="bar c2"
        color="rgb(166, 224, 251)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={16}
        x={0}
        y={0}
      />
      <rect
        className="bar c2"
        color="rgb(166, 224, 251)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={14}
        x={0}
        y={0}
      />
      <rect
        className="bar c2"
        color="rgb(166, 224, 251)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={13}
        x={0}
        y={0}
      />
      <rect
        className="bar c2"
        color="rgb(166, 224, 251)"
        onClick={[Function]}
        size={
          Object {
            "height": 0,
            "width": 0,
          }
        }
        value={13}
        x={0}
        y={0}
      />
    </g>
  </g>
</svg>
`;

exports[`Should render correctly with only lines 1`] = `
.c2 {
  cursor: pointer;
  fill: rgb(255,255,255);
  stroke: rgb(127,22,42);
  stroke-width: 3;
}

.c4 {
  cursor: pointer;
  fill: rgb(255,255,255);
  stroke: rgb(255,121,147);
  stroke-width: 3;
}

.c1 {
  fill: none;
  pointer-events: none;
  stroke: rgb(127,22,42);
  stroke-width: 3;
}

.c3 {
  fill: none;
  pointer-events: none;
  stroke: rgb(255,121,147);
  stroke-width: 3;
}

.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-18"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
    <g
      className="cherries-layer"
    >
      <path
        className="line-path c1"
        color="rgb(127, 22, 42)"
        d="M0,0L0,0L0,0L0,0"
      />
      <g
        className="line-dot-group"
      >
        <circle
          className="line-dot c2"
          color="rgb(127, 22, 42)"
          cx={0}
          cy={0}
          r={5}
        />
        <circle
          className="line-dot c2"
          color="rgb(127, 22, 42)"
          cx={0}
          cy={0}
          r={5}
        />
        <circle
          className="line-dot c2"
          color="rgb(127, 22, 42)"
          cx={0}
          cy={0}
          r={5}
        />
        <circle
          className="line-dot c2"
          color="rgb(127, 22, 42)"
          cx={0}
          cy={0}
          r={5}
        />
      </g>
    </g>
    <g
      className="dates-layer"
    >
      <path
        className="line-path c3"
        color="rgb(255, 121, 147)"
        d="M0,0L0,0L0,0L0,0"
      />
      <g
        className="line-dot-group"
      >
        <circle
          className="line-dot c4"
          color="rgb(255, 121, 147)"
          cx={0}
          cy={0}
          r={5}
        />
        <circle
          className="line-dot c4"
          color="rgb(255, 121, 147)"
          cx={0}
          cy={0}
          r={5}
        />
        <circle
          className="line-dot c4"
          color="rgb(255, 121, 147)"
          cx={0}
          cy={0}
          r={5}
        />
        <circle
          className="line-dot c4"
          color="rgb(255, 121, 147)"
          cx={0}
          cy={0}
          r={5}
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`Should render correctly with width 1`] = `
.c0 {
  height: 90px;
  width: 400px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-2"
  size={
    Object {
      "height": 90,
      "width": 400,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;

exports[`Should render correctly with width and height 1`] = `
.c0 {
  height: 300px;
  width: 400px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-6"
  size={
    Object {
      "height": 300,
      "width": 400,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;

exports[`render correctly with x axis label 1`] = `
.c1 {
  text-anchor: middle;
}

.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-8"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <text
      className="x-axis-label c1"
      height={0}
      width={0}
      x={0}
      y={33.333333333333336}
    >
      foo
    </text>
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;

exports[`render correctly with y axis label 1`] = `
.c1 {
  text-anchor: middle;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c0 {
  height: 90px;
  width: 100px;
}

<svg
  className="silky-charts c0"
  id="silky-charts-bar-line-10"
  size={
    Object {
      "height": 90,
      "width": 100,
    }
  }
>
  <g
    className="silky-charts-container"
    transform="translate(50, 40)"
  >
    <text
      className="y-axis-label c1"
      height={0}
      width={0}
      x={-0}
      y={-33.333333333333336}
    >
      foo
    </text>
    <g
      className="axis-x "
      transform="translate(0, 0)"
    />
    <g
      className="axis-y "
    />
  </g>
</svg>
`;
